include/swift/Runtime/Metadata.h

```diff
@@ -1901,6 +1901,9 @@ struct TargetProtocolRequirement {
   ProtocolRequirementFlags Flags;
   // TODO: name, type
 
+  /// Dispatch thunk. Will be null if the protocol is not resilient.
+  RelativeDirectPointer<void, /*nullable*/ true> Function;
```

Comment:
I think this comment, and probably the field name, could be a lot clearer.  I'll admit that I went down a garden path because of the word "dispatch" in the comment, but still, some sort of statement about what the function does would be good.  It only clicked for me when I read the rest of your patch.

You should include some level of explanation of why this needs to be linked here.  Currently that's just in the commit message.

Is it really a good idea to base this keying off of function-pointer equality?  I'll admit to feeling a bit burnt from C++ inline functions, and some of those reasons don't apply here, but... function pointers are very opaque, and there are sometimes good reasons to introduce thunks around them.  On the other hand, it *is* a symbol that's guaranteed to be exported.